### PR TITLE
Unit tests for Batch File Processing Rocket

### PR DESCRIPTION
**Pull request by charlietfe**
## Description
Added unit test to Batch File Processing Rocket

## Checks
- [X] Project Builds
- [X] Project passes tests and checks
- [N/A] Updated documentation accordingly

### CODE REVIEW

**Review by adriangmweb — APPROVED** [review]
LGTM! 🚀

**Review by davidverdu — APPROVED** [review]
LGTM. The only doubt I have is if we should put the `package-lock.json` files in `.gitignore` in place of adding it in the PR. When I compile the project and launch the tests in my machine, some `package-lock.json` files, including the one added by this PR, get modified and appear to be modified if I run `git status`.

**Comment by javiertoledo** [comment]
There are different opinions on what to do with lock files. In my opinion, having the `package-lock.json` checked in the repository makes sense because it helps us to make sure that everyone is using the same package versions, and if someone upgrades a package, then everyone else can see it in the PR.

File: packages/rocket-batch-file-process-aws-infrastructure/src/lambdas/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        "xmlbuilder": "~9.0.1"
+      }
+    },
+    "xmlbuilder": {
+      "version": "9.0.7",
+      "resolved": "https://registry.npmjs.org/xmlbuilder/-/xmlbuilder-9.0.7.tgz",
+      "integrity": "sha1-Ey7mPS7FVlxVfiD0wi35rKaGsQ0="
+    }
+  }
+}
```
Comment (javiertoledo) [inline packages/rocket-batch-file-process-aws-infrastructure/src/lambdas/package-lock.json:283]: We should find a better way of doing this, having a package inside of a package is... weird 🤔

**Review by javiertoledo — APPROVED** [review]

File: packages/rocket-batch-file-process-aws-infrastructure/test/unit/batch-file-processing.test.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    params = {
+      bucketName,
+      chunkSize,
+    }
+  })
+
+  context('on mount', () => {
+    const verifyFunctionPolicies = (currentPolicies: any[], expectedPolicies: any[]) => {
+      expectedPolicies.forEach((item) => {
+        const plcy = currentPolicies.find((policy) => {
```
Comment (NickSeagull) [inline packages/rocket-batch-file-process-aws-infrastructure/test/unit/batch-file-processing.test.ts:71]: I think it'd be better inverting the names:
```suggestion
        const policy = currentPolicies.find((plcy) => {
```

File: packages/rocket-batch-file-process-aws-infrastructure/test/unit/batch-file-processing.test.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+      const notifications = getConstructObjValueForKey(sourceBucketObj, 'notifications')['lambdaNotifications']
+      expect(notifications).containSubset([{ Events: [EventType.OBJECT_CREATED] }])
+
+      // Stagin Bucket
+      const staginBucket = appStack.node.tryFindChild('stagingUploadBucket') as Bucket
+      expect(staginBucket).not.to.be.null
+
+      const staginBucketObj = constructToObj(staginBucket)
+      expect(staginBucketObj).containSubset([{ key: 'physicalName', value: `${bucketName}-staging` }])
```
Comment (NickSeagull) [inline packages/rocket-batch-file-process-aws-infrastructure/test/unit/batch-file-processing.test.ts:97]: These should be `staging` not `stagin`

**Comment by charlietfe** [comment]
The test will be included in its own rocket repo.

**Pull request closed without merging**